src/ucs/sys/sys.c

```diff
@@ -1549,6 +1548,40 @@ unsigned long ucs_sys_get_proc_create_time(pid_t pid)
     return 0ul;
 }
 
+ucs_status_t ucs_sys_get_memlock_rlimit(rlim_t *rlimit_value)
+{
+    struct rlimit limit_info;
+    int res;
+
+    /* Check the value of the max locked memory which is set on the system
+     * (ulimit -l) */
+    res = getrlimit(RLIMIT_MEMLOCK, &limit_info);
+    if (res == 0) {
+        *rlimit_value = limit_info.rlim_cur;
+        return UCS_OK;
+    } else {
+        ucs_debug("unable to get the max locked memory limit: %m");
+        return UCS_ERR_IO_ERROR;
+    }
```

Comment:
```
if (res != 0) {
      ucs_debug("unable to get the max locked memory limit: %m");
      return UCS_ERR_IO_ERROR;
}

...
```

Reply:
In this case, I got `error: control reaches the end of non-void function` error from the compiler. Should I do something like this:
```C
if (res != 0) {
      ucs_debug("unable to get the max locked memory limit: %m");
      return UCS_ERR_IO_ERROR;
}
return UCS_ERR_IO_ERROR;
```

Or maybe it's better to choose the following option?
```C
if (res == 0) {
    *rlimit_value = limit_info.rlim_cur;
    return UCS_OK;
}
ucs_debug("unable to get the max locked memory limit: %m");
return UCS_ERR_IO_ERROR;
```